app/src/main/java/org/wikipedia/page/references/References.java

```diff
@@ -0,0 +1,64 @@
+package org.wikipedia.page.references;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.annotations.SerializedName;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+
+public class References {
+    @SuppressWarnings("unused,NullableProblems")  @NonNull private String revision;
+    @SuppressWarnings("unused,NullableProblems") @NonNull private String tid;
+    @SuppressWarnings("unused,NullableProblems") @NonNull @SerializedName("references_by_id") private Map<String, Reference> referencesMap;
```

Comment:
Marking things as `@NonNull` does not guarantee that they will actually be NonNull when it's deserialized. Everything should be `@Nullable`, and then checked/guarded in the getter methods.

Reply:
Done!